src/pages/HomeView.vue

```diff
@@ -27,12 +27,112 @@ import {
   TableRow,
 } from '@/components/ui/table'
 import Separator from '@/components/ui/separator/Separator.vue'
+import * as CryptoJS from 'crypto-js';
+import * as secp256k1 from 'secp256k1';
 
 const isPaid = ref(false)
 const isVerified = ref(false)
 
 const { form } = useForm()
 
+const bech32CharValues = 'qpzry9x8gf2tvdw0s3jn54khce6mua7l';
+
+const bech32To8BitArray = (str: string) => {
+    const int5Array = str.split('').map(char => bech32CharValues.indexOf(char));
+    
+    let count = 0;
+    let buffer = 0;
+    const byteArray = [];
+    
+    int5Array.forEach((value) => {
+        buffer = (buffer << 5) + value;
+        count += 5;
+        
+        while (count >= 8) {
+            byteArray.push(buffer >> (count - 8) & 255);
+            count -= 8;
+        }
+    });
+    
+    if (count > 0) {
+        byteArray.push(buffer << (8 - count) & 255);
+    }
+
+    return Uint8Array.from(byteArray);
+}
+
+const strToHex = (str: string) => {
+  return str.split('').map(x => x.charCodeAt(0).toString(16)).join('');
+}
+
+const wordListToUint8Array = (wordList: CryptoJS.lib.WordArray) => {
+  const dataArray = new Uint8Array(wordList.sigBytes);
+
+  for (let i = 0x0; i < wordList.sigBytes; i++) {
+    dataArray[i] = wordList.words[i >>> 0x2] >>> 0x18 - i % 0x4 * 0x8 & 0xff;
+  }
+  
+  const data = new Uint8Array(dataArray);
+
+  return data;
+}
+
+const byteArrayToHexString = (byteArray: Uint8Array) => {
+  return Array.prototype.map.call(byteArray, function (byte) {
+      return ('0' + (byte & 0xFF).toString(16)).slice(-2);
+  }).join('');
+}
+
+const getPubkeyFromSignature = (decoded: bolt11.DecodedInvoice) => {
+  const signature = decoded.sections.find(section => section.name === 'signature');
+  
+  if (!signature || !signature.letters || !signature.value) {
+    return null;
+  }
+
+  const prefixSections = ['lightning_network', 'coin_network', 'amount'];
+
+  const prefix = decoded.sections
+    .filter(section => prefixSections.includes(section.name))
+    .map(section => {
+      if ('letters' in section) return section.letters
+    })
+    .join('');
+
+  if (!prefix) {
+    return null;
+  }
+
+  const separator = decoded.sections.find(section => section.name === 'separator')?.letters;
+
+  if (!separator) {
+    return null;
+  }
+
+  const splitInvoice = decoded.paymentRequest.split(prefix + separator);
+
+  const data = splitInvoice[1].split(signature.letters)[0];
+  
+  const signingData = 
+    strToHex(prefix) + 
+    byteArrayToHexString(bech32To8BitArray(data));
+
+  const signingDataParsed = CryptoJS.enc.Hex.parse(signingData);
+  const payHash = CryptoJS.SHA256(signingDataParsed);
```

Comment:
I don't think you need this CryptoJS lib here. You could use `crypto.subtle.digest('SHA-256', ...)` function here which is native from the browser